e3sm_diags/driver/zonal_mean_xy_driver.py

```diff
@@ -201,129 +205,83 @@ def _run_diags_3d(
-                parameter,
-                ds_test_ilev,
-                ds_ref_ilev,
-                var_key,
-                region,
-            )
+            logger.info(f"Selected region: {region}")
+
             parameter._set_param_output_attrs(var_key, season, region, ref_name, ilev)
-            metrics_dict = {}
-            da_diff_1d = regrid_to_lower_res_1d(da_test_1d, da_ref_1d)
+
+            da_test_1d, da_ref_1d = _calc_zonal_mean(ds_test_ilev, ds_ref_ilev, var_key)
+            da_diff_1d = _get_diff_of_zonal_means(da_test_1d, da_ref_1d)
+
             _save_data_metrics_and_plots(
                 parameter,
                 plot_func,
                 var_key,
                 da_test_1d.to_dataset(),
                 da_ref_1d.to_dataset(),
                 da_diff_1d.to_dataset(),
-                metrics_dict,
+                metrics_dict=None,
             )
 
 
-def _calculate_zonal_mean(
-    parameter: CoreParameter,
+def _calc_zonal_mean(
     ds_test: xr.Dataset,
     ds_ref: xr.Dataset,
     var_key: str,
-    region: str,
-) -> Tuple[xr.Dataset, xr.Dataset | None]:
-    """Get zonal mean metrics and save data (optional), metrics, and plots
+) -> Tuple[xr.DataArray, xr.DataArray]:
+    """Calculate zonal mean metrics.
+
+    # TODO: Write unit tests for this function.
 
     Parameters
     ----------
-    parameter : CoreParameter
-        The parameter for the diagnostic.
     ds_test : xr.Dataset
         The dataset containing the test variable.
     ds_ref : xr.Dataset
         The dataset containing the ref variable. If this is a model-only run
         then it will be the same dataset as ``ds_test``.
     var_key : str
         The key of the variable.
-    region : str
-        The region.
 
     Returns
     -------
-    [xr.Dataset, xr.Dataset | None]
-        A Tuple containing test dataset, the ref dataset (optional)
-
-    Raises
-    ------
-    RuntimeError
-        If none "global" region is supplied.
+    Tuple[xr.DataArray, xr.DataArray]
+        A Tuple containing the zonal mean for the test variable and the ref
+        variable.
     """
-    logger.info(f"Selected region: {region}")
-    parameter.var_region = region
+    da_test_1d = ds_test.spatial.average(var_key, axis="X", weights="generate")[var_key]
+    da_ref_1d = ds_ref.spatial.average(var_key, axis="X", weights="generate")[var_key]
 
-    if region != "global":
-        raise RuntimeError(
-            "None global region is not currently supported for zonal mean xy set. Aborting."
-        )
+    return da_test_1d, da_ref_1d
 
-    # Align the grid resolutions if the diagnostic is not model only.
-    if not parameter.model_only:
-        ds_test_1d = zonal_avg(ds_test, var_key)
-        ds_ref_1d = zonal_avg(ds_ref, var_key)
-    else:
-        ds_test_1d = zonal_avg(ds_test, var_key)
-        ds_ref_1d = None
 
-    return ds_test_1d, ds_ref_1d  # type: ignore
```

Comment:
`_get_diff_of_zonal_means()` and the plotter function in `zonal_mean_xy_plot.py` always expect a reference dataset, so we can't `ds_ref_1d =  None` (mypy warnings). 

Instead, always take the spatial average of test and ref datasets.